internal/jobservice/eventstojobs/eventstojobs.go

```diff
@@ -60,42 +69,51 @@ func (eventToJobService *EventsToJobService) streamCommon(clientConnect *client.
 		ticker := time.NewTicker(time.Duration(eventToJobService.jobServiceConfig.SubscribeJobSetTime) * time.Second)
 		for range ticker.C {
 			if !eventToJobService.jobServiceRepository.IsJobSetSubscribed(eventToJobService.queue, eventToJobService.jobSetId) {
-				return conn.Close()
+				cancel()
+				return nil
 			}
 		}
 		return nil
 	})
 	g.Go(func() error {
-		select {
-		case <-ctx.Done():
-			return ctx.Err()
-		default:
-		}
-		eventClient := api.NewEventClient(conn)
-		stream, err := eventClient.GetJobSetEvents(ctx, &api.JobSetRequest{
-			Id:             eventToJobService.jobSetId,
-			Queue:          eventToJobService.queue,
-			Watch:          true,
-			FromMessageId:  fromMessageId,
-			ErrorIfMissing: false,
-		})
-		if err != nil {
-			log.Errorf("Error found from client %v", err)
-			return err
-		}
+		// this loop will run until the context is canceled or an error is encountered
 		for {
-
-			msg, err := stream.Recv()
-			if err != nil {
-				log.Error(err)
-				return err
+			conn, connErr := client.CreateApiConnection(clientConnect)
```

Comment:
I don't think we want this.  A big point of this was to only have a single connection for a given job-set.

We also want to infinitely listen on the stream of the GetJobSetEvents rather than keep calling it.